### PR TITLE
hide registration

### PR DESCRIPTION
**Pull request by AndreyLykhoman**
**Description:** Hide registration for  students

**Youtrack:** https://youtrack.raccoongang.com/issue/DIDACTICUM-28

### CODE REVIEW

File: common/djangoapps/student/views.py [hunk 1]
```diff
@@ -470,6 +470,10 @@ def signin_user(request):
 def register_user(request, extra_context=None):
     """Deprecated. To be replaced by :class:`student_account.views.login_and_registration_form`."""
     # Determine the URL to redirect to following login:
+    if 'ALLOW_PUBLIC_ACCOUNT_CREATION' in settings.FEATURES and \
```
Comment (flying-pi) [inline common/djangoapps/student/views.py]: Please use get method with a default value

File: common/djangoapps/student/views.py [hunk 2]
```diff
@@ -470,6 +470,10 @@ def signin_user(request):
...
+            settings.FEATURES["ALLOW_PUBLIC_ACCOUNT_CREATION"]:
+        return Http404('<h1>Page not found</h1>')
```
Comment (flying-pi) [inline common/djangoapps/student/views.py]: You can use compact variant `raise Http404`

File: lms/djangoapps/student_account/views.py [hunk 3]
```diff
@@ -67,6 +67,10 @@ def login_and_registration_form(request, initial_mode="login"):
         initial_mode (string): Either "login" or "register".
 
     """
+    if initial_mode == "register" and \
+            'ALLOW_PUBLIC_ACCOUNT_CREATION' in settings.FEATURES and \
```
Comment (flying-pi) [inline lms/djangoapps/student_account/views.py]: Please use get method with a default value and avoid slash for the long string.

File: lms/djangoapps/student_account/views.py [hunk 4]
```diff
@@ -67,6 +67,10 @@ def login_and_registration_form(request, initial_mode="login"):
...
+            settings.FEATURES["ALLOW_PUBLIC_ACCOUNT_CREATION"]:
+        return HttpResponseNotFound('<h1>Page not found</h1>')
```
Comment (flying-pi) [inline lms/djangoapps/student_account/views.py]: You can use compact variant raise Http404

File: common/djangoapps/student/views.py (hunk 1 above)
Comment (AndreyLykhoman) [reply common/djangoapps/student/views.py]: Done

File: common/djangoapps/student/views.py (hunk 2 above)
Comment (AndreyLykhoman) [reply common/djangoapps/student/views.py]: Done

File: lms/djangoapps/student_account/views.py (hunk 3 above)
Comment (AndreyLykhoman) [reply lms/djangoapps/student_account/views.py]: Done

File: lms/djangoapps/student_account/views.py (hunk 4 above)
Comment (AndreyLykhoman) [reply lms/djangoapps/student_account/views.py]: Done

**Pull request merged**